src/main/java/hu/oe/nik/szfmv/automatedcar/math/IVector.java

```diff
@@ -0,0 +1,184 @@
+     * Rotation is mathematical, positive is towards the left.*/
+    default double getDegreesRelativeTo(Axis axis) {
+        return getRadiansRelativeTo(axis.positiveDirection()) / PI * 180;
+    }
+
+    /**Gets the angle of the vector in degrees relative to the given direction.
+     * Rotation is mathematical, positive is towards the left.
+     * @return Value from -180° to +180°.*/
+    default double getDegreesRelativeTo(IVector direction) {
+        return getRadiansRelativeTo(direction) / PI * 180;
+    }
+
+    /**Gets angle of vector in absolute radians relative to the given axis.
+     * Rotation is mathematical, positive is towards the left.
+     * @return Value between 0 rad and 2*PI rad.*/
+    default double getAbsRadiansRelativeTo(Axis axis) {
+        double signedRads = getRadiansRelativeTo(axis);
+        return (signedRads < 0)
+                ? (2*PI + signedRads)
+                : signedRads;
+    }
+
+    /**Gets angle of vector in absolute radians relative to the given direction.
+     * Rotation is mathematical, positive is towards the left.
+     * @return Value between 0 rad and 2*PI rad.*/
+    default double getAbsRadiansRelativeTo(IVector direction) {
+        double signedRads = getRadiansRelativeTo(direction);
+        return (signedRads < 0)
+                ? (2*PI + signedRads)
+                : signedRads;
+    }
+
+    /**Gets angle of vector in absolute degrees relative to the given axis.
+     * Rotation is mathematical, positive is towards the left.
+     * @return Value between 0° and 360°.*/
+    default double getAbsDegreesRelativeTo(Axis axis) {
+        return getAbsRadiansRelativeTo(axis.positiveDirection()) / PI * 180;
+    }
+
+    /**Gets angle of vector in absolute degrees relative to the given direction.
+     * Rotation is mathematical, positive is towards the left.
+     * @return Value between 0° and 360°.*/
+    default double getAbsDegreesRelativeTo(IVector direction) {
+        return getAbsRadiansRelativeTo(direction) / PI * 180;
+    }
+
+    /**Creates a new vector with the given X and Y differences, which also define the length and angle of the vector.
+     * @see Axis#positiveDirection()
+     * @see Axis#negativeDirection() */
+    static IVector vectorFromXY(double x, double y) {
+        return new IVector() {
+
+            @Override
+            public double getXDiff() {
+                return x;
+            }
+
+            @Override
+            public double getYDiff() {
+                return y;
+            }
+
+            @Override public double getLength() {
+                return sqrt(x*x + y*y);
+            }
+
+            @Override public double getRadiansRelativeTo(IVector direction) {
+                double xRads = direction.getRadiansRelativeTo(Axis.X);
+                double relativeRads  = (getRadiansRelativeToAxisX() - xRads) % (2 * PI);
+                while (relativeRads < -PI) relativeRads += 2*PI;
```

Comment:
Hol van az előírt blokkjelölés?

Reply:
ezek mergaből jöttek, de köszi

Reply:
Esetleg kirakhatom külön metódusba, pl.:
```java
//...
double relativeRads  = (getRadiansRelativeToAxisX() - xRads) % (2 * PI);
while (relativeRads < -PI) relativeRads += 2*PI;
//...
```
helyett:
```java
//...
double relativeRads  = inFirstRadianPeriod(getRadiansRelativeToAxisX() - xRads));
//...

private static double inFirstRadianPeriod(double radians) {
    double radians = radians % (2 * PI);
    while (relativeRads < -PI) {
        relativeRads += 2*PI;
    }
    return radians;
}
```

Reply:
@AbyssPrince  Tudom, hogy ott kellett volna ezt észrevennem, ebben teljesen igazad van, ahogy az issue-ban is írtam.

Az észrevételen kizárólag ennek szólt:

```java
while (relativeRads < -PI) {
    relativeRads += 2*PI;
}
```
Ez az előírt kódformázás blokkokra vonatkozó része. Nem gondolom, hogy ezt külön metódusba kellene szervezni.